### PR TITLE
[FLOC-2177] Add EMC links to Install doc

### PR DESCRIPTION
**Pull request by davidoliverSP2**
This is a somewhat light effort, but covers the EMC GitHub repos. I discussed this @ferrantim and this covers the EMC backend, but we will need to expand this in the future.

I've created FLOC 2442 and 2443 to make this whole section easier to find, and use.

### CODE REVIEW

File: docs/using/installing/index.rst [hunk 1]
```diff
@@ -559,6 +559,19 @@ The configuration item to use AWS should look like:
...
+EMC Block Device Backend Configuration
+......................................
+
+EMC provide plugins for Flocker integration with `ScaleIO`_ and `XtremIO`_.
+For more information, including installation, testing and usage instructions, see the following links to their GitHub repositories:
+
+* `EMC ScaleIO Flocker driver on GitHub`_
+* `EMC XtremIO Flocker driver on GitHub`_
+
+.. XXX FLOC xxx to improve this EMC/Backend storage section
```
Comment (adamtheturtle) [inline docs/using/installing/index.rst]: Link to the follow up

Comment (adamtheturtle) [inline docs/using/installing/index.rst]: * as in, replace xxx with the number

File: docs/using/installing/index.rst [hunk 2]
```diff
@@ -559,6 +559,19 @@ The configuration item to use AWS should look like:
 Make sure that the ``region`` and ``zone`` match each other and that both match the region and zone where the Flocker agent nodes run.
 AWS must be able to attach volumes created in that availability zone to your Flocker nodes.
 
+.. _emc-dataset-backend:
```
Comment (adamtheturtle) [inline docs/using/installing/index.rst]: Don't add unused references. However I think that these should be linked to by the Release Notes sentence which mentions them.

**Comment by adamtheturtle** [comment]
The installation page is huge and unwieldy but I think that that is a problem for another day.

File: docs/using/installing/index.rst (hunk 1 above)
Comment (davidoliverSP2) [inline docs/using/installing/index.rst]: I have on my local version, maybe I missed a commit before pushing

File: docs/using/installing/index.rst [hunk 3]
```diff
@@ -559,6 +559,19 @@ The configuration item to use AWS should look like:
...
+
+EMC Block Device Backend Configuration
+......................................
+
+EMC provide plugins for Flocker integration with `ScaleIO`_ and `XtremIO`_.
+For more information, including installation, testing and usage instructions, see the following links to their GitHub repositories:
```
Comment (adamtheturtle) [inline docs/using/installing/index.rst]: :question: I'm not sure what it means to "see" a link. Perhaps "visit".

**Comment by adamtheturtle** [comment]
Link to these where it says "Examples of such storage include Amazon EBS and EMC ScaleIO." in the "Dataset Backend Plugins" page.

**Comment by adamtheturtle** [comment]
Thanks @davidoliverSP2 please address and resubmit.

File: docs/using/installing/index.rst [hunk 4]
```diff
@@ -678,3 +691,8 @@ Next Step
 ---------
 
 The next section describes your next step - setting up an :ref:`authenticated user<authenticate>`.
+
+.. _ScaleIO: http://www.emc.com/storage/scaleio/index.htm
+.. _XtremIO: http://www.emc.com/storage/xtremio/overview.htm
```
Comment (exarkun) [inline docs/using/installing/index.rst]: We should link to `https` resources whenever possible.

**Comment by davidoliverSP2** [comment]
@adamtheturtle There are currently quite a few Jira issues related to improving our Install documentation:

2442, 2443, 2403, 2339, 2266, 2212, 2128, 2077, 1975

I'm collating these into a story which will coordinate all requested improvements, and design a new layout/structure for the Install docs. This is still in pre-design at present though

File: docs/gettinginvolved/plugins.rst [hunk 5]
```diff
@@ -11,7 +11,7 @@ Block device backends
 =====================
 
 Flocker implements generic logic for network-based block device storage.
-Examples of such storage include Amazon EBS and EMC ScaleIO.
+Examples of such storage include :ref:`Amazon EBS<aws-dataset-backend>` and :ref:`EMC ScaleIO/XtremIO<emc-dataset-backend>`.
```
Comment (adamtheturtle) [inline docs/gettinginvolved/plugins.rst]: This makes it seem like ScaleIO and XtremIO are just different names for the same thing. Maybe this needs an "and".

**Comment by adamtheturtle** [comment]
Thanks, the release notes should still be changed (as per https://github.com/ClusterHQ/flocker/pull/1677#discussion_r32736880) and the "/" thing but otherwise this is good to be merged as long as the docs builder passes.

**Pull request merged**